thetis/inversion_tools.py

```diff
@@ -182,21 +185,29 @@ def register_observation_data(self, station_names, variable, time,
         :arg list values: array of observations, one for each station
         :arg list x: list of station x coordinates
         :arg list y: list of station y coordinates
+        :kwarg list start_times: optional start times for the observation periods
+        :kwarg list end_times: optional end times for the observation periods
         """
         self.station_names = station_names
+        num_stations = len(station_names)
         self.variable = variable
         self.observation_time = time
         self.observation_values = values
         self.observation_x = x
         self.observation_y = y
+        start_times = start_times or -numpy.ones(num_stations)*numpy.inf
+        self.obs_start_times = numpy.array(start_times)
+        end_times = end_times or numpy.ones(num_stations)*numpy.inf
+        self.obs_end_times = numpy.array(end_times)
```

Comment:
Apologies for repeatedly pushing to the PR. I had got this bit wrong (nan vs inf!) and it was breaking things in the `channel_inversion` example.